### PR TITLE
SLES15 enablement

### LINKED ISSUE #1733: Add SLES15 to the e2e-testing matrix of linux vms we test against
We would like to cite official support for this and so manual testing is done and has revealed no immediate risk.  The plan is to add this to the e2e-testing framework hosts list for Agent tests.

in support of this ticket:
https://github.com/elastic/beats/issues/27230

### PR DESCRIPTION
**Pull request by jlind23**
Closes #1733

### CODE REVIEW

File: .ci/.e2e-tests.yaml [hunk 1]
```diff
@@ -35,7 +38,7 @@ SUITES:
     scenarios:
       - name: "Fleet"
         tags: "fleet_mode_agent"
-        platforms: ["fleet_arm64", "fleet_amd64"]
+        platforms: ["fleet_arm64", "fleet_amd64","fleet_sles15"]
```
Comment (v1v) [inline .ci/.e2e-tests.yaml:41]: nit:

```suggestion
        platforms: ["fleet_arm64", "fleet_amd64", "fleet_sles15"]
```

File: .ci/ansible/vars/SLES.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+ansible_user: root
```
Comment (adam-stokes) [inline .ci/ansible/vars/SLES.yml:1]:
```suggestion
ansible_user: ec2-user
```

Comment (adam-stokes) [reply .ci/ansible/vars/SLES.yml:1]: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/connection-prereqs.html

File: .ci/ansible/playbook.yml [hunk 3]
```diff
@@ -93,11 +93,16 @@
...
-    ansible_user: admin
     ansible_python_interpreter: /usr/bin/python3
   pre_tasks:
     - apt: {update_cache: yes}
   tasks:
+  - include_vars: '{{ ansible_distribution }}.yml'
+
+  - name: Print Ansible User obtained from the distribution vars
+    ansible.builtin.debug:
+    var: "ansible_user"
```
Comment (v1v) [inline .ci/ansible/playbook.yml:104]: What do you think this?

```suggestion
  - name: Print Ansible User obtained from the distribution vars
    ansible.builtin.debug:
      msg: ansible_user {{ ansible_user }}
```

or

```suggestion
  - name: Print Ansible User obtained from the distribution vars
    ansible.builtin.debug:
      var: ansible_user
```

from https://docs.ansible.com/ansible/latest/collections/ansible/builtin/debug_module.html#examples

**Comment by adam-stokes** [comment]
Also, we need to pull the ansible_user var into this as well:

https://github.com/elastic/e2e-testing/blob/main/.ci/ansible/tasks/runners.yml#L39

**Comment by jlind23** [comment]
@mdelapenya will you take care of this?

**Comment by mdelapenya** [comment]
Yes, assigning to me

**Comment by adam-stokes** [comment]
Ok, this should do it, the last error was inside the Jenkinsfile wrt directory permission, but the actual tests on SLES itself passed.

**Comment by jlind23** [comment]
@adam-stokes if the test succeed can we merge it then?

**Review by mdelapenya — APPROVED** [review]
Awesome! LGTM

image

**Comment by mdelapenya** [comment]
> @adam-stokes if the test succeed can we merge it then?

@jlind23 I'm going to merge this little one, as the SLES tests passed. The only test error is related to k8s-autodiscover and the elastic-agent test suite, which is something reported in #1992

File: .ci/.e2e-tests-for-elastic-agent.yaml [hunk 4]
```diff
@@ -3,21 +3,31 @@ PLATFORMS:
   stack:
     image: "ami-0d90bed76900e679a"
     instance_type: "c5.4xlarge"
+    username: admin
```
Comment (mdelapenya) [inline .ci/.e2e-tests-for-elastic-agent.yaml:6]: Well done here! As this is the descriptor of the machine, it makes total sense to have the configuration here

**Pull request merged**